src/librustdoc/clean/mod.rs

```diff
@@ -1303,8 +1310,13 @@ fn resolution_failure(
         }
         diag
     } else {
-        cx.sess().struct_span_warn(sp, &msg)
+        cx.tcx.struct_span_lint_node(lint::builtin::INTRA_LINK_RESOLUTION_FAILURE,
+                                     NodeId::new(0),
+                                     sp,
+                                     &msg)
     };
+    diag.help("to escape `[` and `]` characters, either put them into \"`[]`\" or \
+               use HTML values `&#91;` and `&#93;`");
```

Comment:
I'd write

> put them between back ticks (like "`[]`") or

Also, doesn't backslash escaping work, too?

Reply:
It totally does haha. Just writing this then!